scipy/signal/_spectral_py.py

```diff
@@ -1772,13 +1786,15 @@ def _spectral_helper(x, y, fs=1.0, window='hann', nperseg=None, noverlap=None,
 
     axis = int(axis)
 
-    # Ensure we have np.arrays, get outdtype
-    x = np.asarray(x)
+    # Ensure we have xp.arrays, get outdtype
+    x = xp.asarray(x)
+    # https://github.com/data-apis/array-api-compat/issues/43
```

Comment:
This should be able to be cleaned up now after https://github.com/data-apis/array-api-compat/pull/55.